### ISSUE TITLE
X360CE/HIDGuardian won't hide original controller

### ISSUE THREAD

**Issue by yousie642** [x360ce/x360ce#1201]
I have HIDGuardian installed, and the "Hide" box is checked in the device list, but both the original DInput controller and the virtual 360 controller continue to appear in the Control Panel Game Controllers list and in games.

**Comment by AlexMrBison** [comment]
Find ur device id then add to this 
Right click on ur joystick > properties > HID > Details > hardware Ids  then copy the 3 fisrt lines.

and add here>
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\HidGuardian\Parameters
"Example my device ids"
"AffectedDevices"=
HID\VID_7545&PID_1122&REV_0100
HID\VID_7545&PID_1122
HID\VID_7545&UP:0001_U:0004

More info here > https://github.com/csutorasa/XOutput/blob/master/README.md

**Comment by andreonlive** [comment]
Guys I have problems with HIDGuardian not hiding Dinputs as well, I really would like to find a way to fix it.

I have 03 Ipega PG-9139 (Bluetooth) and everything is perfect EXCEPT there is no way to hide Direct Input.

Look I made some screenshots of my settings, below there's just one controller connected.
Please guy... what to do to hide it?

![img 01](https://user-images.githubusercontent.com/74944196/100042857-0e7ced00-2e58-11eb-8552-e51e6c77c80c.gif)
![img 02 0](https://user-images.githubusercontent.com/74944196/100042871-1472ce00-2e58-11eb-8c55-0b7c0669d171.gif)
![img 02 1](https://user-images.githubusercontent.com/74944196/100042888-19378200-2e58-11eb-9e7d-bdfd78d11eab.gif)
![img 02 2](https://user-images.githubusercontent.com/74944196/100042894-1ccb0900-2e58-11eb-8742-b1dbf36b5b10.gif)
![img 03](https://user-images.githubusercontent.com/74944196/100042901-1fc5f980-2e58-11eb-9e99-897799e30316.gif)
![img 05](https://user-images.githubusercontent.com/74944196/100042923-25bbda80-2e58-11eb-9df8-84c5b264ba46.gif)
![img 06](https://user-images.githubusercontent.com/74944196/100042928-281e3480-2e58-11eb-9edb-b2821bbf51c0.gif)

**Comment by FoxBearBear** [comment]
> Guys I have problems with HIDGuardian not hiding Dinputs as well, I really would like to find a way to fix it.
> 
> I have 03 Ipega PG-9139 (Bluetooth) and everything is perfect EXCEPT there is no way to hide Direct Input.
> 
> Look I made some screenshots of my settings, below there's just one controller connected.
> Please guy... what to do to hide it?
> 
> ![img 01](https://user-images.githubusercontent.com/74944196/100042857-0e7ced00-2e58-11eb-8552-e51e6c77c80c.gif)
> ![img 02 0](https://user-images.githubusercontent.com/74944196/100042871-1472ce00-2e58-11eb-8c55-0b7c0669d171.gif)
> ![img 02 1](https://user-images.githubusercontent.com/74944196/100042888-19378200-2e58-11eb-9e7d-bdfd78d11eab.gif)
> ![img 02 2](https://user-images.githubusercontent.com/74944196/100042894-1ccb0900-2e58-11eb-8742-b1dbf36b5b10.gif)
> ![img 03](https://user-images.githubusercontent.com/74944196/100042901-1fc5f980-2e58-11eb-9e99-897799e30316.gif)
> ![img 05](https://user-images.githubusercontent.com/74944196/100042923-25bbda80-2e58-11eb-9df8-84c5b264ba46.gif)
> ![img 06](https://user-images.githubusercontent.com/74944196/100042928-281e3480-2e58-11eb-9edb-b2821bbf51c0.gif)

Did you get it to work?

**Comment by andreonlive** [comment]
> Did you get it to work?

To be honest YES! I got it after a long research! I'll make a Tutorial today and publish it here for everybody. Than I will try to make video on YouTube about how. Just wait a bit, please

**Comment by andreonlive** [comment]
> Did you get it to work?

Finally I finished the tutorial with many pictures and here it is:

**bit.ly/GamepadTutorial**     (DON'T CLICK over the link, copy it and paste it on your URL bar)



I hope you anjoy!

**Comment by Welis75** [comment]
@andreonlive Can you send me this tutorial please? I really need it because the link you sent is no longer working

**Comment by andreonlive** [comment]
Sure!

Here is the original link from Google Drive:
https://docs.google.com/presentation/d/1hzP_D4jT_rJkmKdajvyKQvH5ozIE1wdW9JBaHnyay7c/edit?usp=sharing

There you can download it.

- - -
*André Franco*
+7 (964) 431-3102


Em qua., 16 de dez. de 2020 às 04:13, Welis75 <notifications@github.com>
escreveu:

**Comment by andreonlive** [comment]
> @andreonlive Can you send me this tutorial please? I really need it because the link you sent is no longer working

DON'T CLICK over the link, copy it and paste it on your URL bar

**Comment by wonx** [comment]
I tried following the tutorial, but in the AutoWhiteLister, when I click on Install HidCerberus, I get an "Operation Failed" error.

It would be great if we could hide the original controller from x360ce itself.

**Comment by andreonlive** [comment]
> I tried following the tutorial, but in the AutoWhiteLister, when I click on Install HidCerberus, I get an "Operation Failed" error.
>

About installing HidCerberus unfortunately idk what to do about... =/
I made this tutorial supposing every step would work naturally.

>
> It would be great if we could hide the original controller from x360ce itself.

About hide using just x360ce, well... Actually it was the original idea.
You can check on "x360ce > Option Tab > HidGuardian Tab" you can install it. 

And on "x360ce > Devices Tab" you should be able to hide every device, so try it.
Unfortunately this way didn't work for me so I had to look for another way and I found WhiteKnight.

But to find this second way I made a LONG research over many web pages to finally find a second way.
My advise for you is try to do the same what I did:
1 - Look for a way to solve your problem with HidCerberus (maybe on their own page - check tutorial)
2 - Look for a way to make x360ce hide it
or
3 - Look for a new way to hide your controllers

I really did my best, I'm sure if you want you will be able to do something as well

Good luck 👍

**Comment by wonx** [comment]
Yep, hiding the devices from within x360ce didn't work for me either, that's why I came here in the first place. I hope it will be fixed in the future.

Anyway, I don't understand why Windows10 does not include a tool to remap an existing controller. Mine has some keys switched (A is Select and B is right trigger, for instance), and it's completely unusable if the game does not support changing the default buttons.

**Comment by andreonlive** [comment]
> Anyway, I don't understand why Windows10 does not include a tool to remap an existing controller. Mine has some keys switched (A is Select and B is right trigger, for instance), and it's completely unusable if the game does not support changing the default buttons.

They do like that because their own controller full compatible with their system.
Basically... They already have what you want, you just have to buy their controller, done.

Generic controllers are made just to work! Not to work well!
If you want a generic controller which works perfectly you will pay for it...
There controller from good brands that simulate an xbox 360 gamepad...
Thats all...

**Comment by Andregate** [comment]
> 
> 
> > Did you get it to work?
> 
> Finally I finished the tutorial with many pictures and here it is:
> 
> **bit.ly/GamepadTutorial** (DON'T CLICK over the link, copy it and paste it on your URL bar)
> 
> I hope you anjoy!

Thanks dude. i never seen someone made tutorial in this well-presented manner. you are legend. hopefully next 360ce can hide device better

**Comment by sehrasaab** [comment]
hi bro i have problem to hide bluetooth controller

i also use white knight but it is not showing my controller name in Available USB/Bluetooth/HID devices
can you fix it?
and my bluetooth controller hardware id is BTHLE\Dev_e0f848010000
plzz find any solution

**Comment by HVR88** [comment]
Is anyone still developing X360CE?  HIDGuardian should be replaced with the original author's new stable project, HIDHide. https://github.com/ViGEm/HidHide

**Comment by EJocys** [comment]
@HVR88
Yes it is in developing. Just that due to circumstances, I had to move more focus onto the primary job. Don't want to burnout :). I will see how to replace HIDGuardian with HIDHide.

**Comment by mauriciabad** [comment]
This issue makes the program totally unusable

**Comment by mauriciabad** [comment]
New workaround: https://github.com/x360ce/x360ce/issues/1389#issue-1415456614

**Comment by PepperWithMint** [comment]
> > Did you get it to work?
> 
> Finally I finished the tutorial with many pictures and here it is:
> 
> **bit.ly/GamepadTutorial** (DON'T CLICK over the link, copy it and paste it on your URL bar)
> 
> I hope you anjoy!

Wow this really helps even after three years. Thank you!

**Comment by hifihedgehog** [comment]
https://github.com/hifihedgehog/PadForge

**Comment by EvoJocys** [comment]
This is a HID Guardian problem. HID Guardian was discontinued by its author in 2023 and x360ce 4.17.50 marked it obsolete; keyboards, mice, audio devices and USB devices dropping out while it was installed were its known side effects. Uninstall it from Options → HID Guardian → Uninstall.

To hide a real controller from games, use HidHide instead: Options → HID Hide shows the driver state and opens its client; there, add x360ce.exe to the application list and tick the controller under Devices. Doing this from inside x360ce is tracked in #1389.

Closing as part of a tracker clean-up after the 4.21 release.
